File: leetcode/invert-binary-tree.java
// https://leetcode.com/problems/invert-binary-tree/

class Solution {
	// Provided TreeNode class
	public class TreeNode {
		int val;
		TreeNode left;
		TreeNode right;
		TreeNode() {}
		TreeNode(int val) { this.val = val; }
		TreeNode(int val, TreeNode left, TreeNode right) {
			this.val = val;
			this.left = left;
			this.right = right;
		}
	}

    public TreeNode invertTree(TreeNode root) {
        if(root != null && (root.left != null || root.right != null)){
            TreeNode newRight = root.left;
            root.left = root.right;
            root.right = newRight;
            invertTree(root.left);
            invertTree(root.right);
        }
        return root;
    }
}

File: leetcode/invert_binary_tree.py
# https://leetcode.com/problems/invert-binary-tree/description/

# Provided TreeNode class
from typing import Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        self.invert(root)
        return root

    def invert(self, node: Optional[TreeNode]) -> None:
        if node is not None:
            temp: Optional[TreeNode] = node.left
            node.left = node.right
            node.right = temp
            self.invert(node.left)
            self.invert(node.right)
